### ISSUE TITLE
[Feature Request] Waking switch from sleep mode

### ISSUE THREAD

**Issue by nsliders** [kenesu-h/sys-hidplus-client-rs#3]
Hi there! Thank you for this awesome alt client. Really like that the home button can be used. 

Request:
Is it possible for the client/sysmodule to be able to wake the console up from sleep using the home button like the joycon/pro controllers do? 

Use case:
Using parsec to play my switch remotely. Currently already have the capture card. However, there is no way to wake the console up to start playing. Waking with the home button would be ideal.

Thanks again!

**Comment by kenesu-h** [comment]
Hey nsliders, thanks for the feature request! This is a feature I've actually been thinking of adding too, but unfortunately there's a caveat or two. Sorry for being a bit long-winded here.

I'm not as familiar with the sysmodule side of things so I'm not totally sure of what libnx (the underlying library behind the sysmodule) is capable of. However, I've done a bit of research by looking into its source code and documentation, and it doesn't seem like there are any functions provided that can wake up the Switch. The only things related to sleep in the source involve the states for the sleep button (presumably the power button?). It doesn't look like much can be done with them, but don't take my word for it.

I've recently taken a look at similar projects to see if they've implemented this - namely MissionControl and sys-con. MissionControl states that non-Switch controllers can't be used to wake up the Switch and lists this as a known issue/limitation. sys-con doesn't seem to mention much about this, but there's one closed issue that briefly mentioned it, albeit with no solution. There was a GBATemp thread or two that wanted similar functionality, though they didn't get far either.

In short, it doesn't look like it's possible - at least right now - to implement this feature, sorry. I will, however, try to keep an eye out for any relevant information in the future. I'll also keep this issue open in case I (or anyone else) have any updates.

**Comment by nsliders** [comment]
Appreciate the detailed response! 

It's a shame it's not easily available. I'll keep a lookout for potential solutions too :)

**Comment by Jayinem** [comment]
Sorry to bump this old request but I still use Client-RS and I was wanting wake from sleep too. I also use Chiaki which is controller/stream PS4 and it uses the same button to turn off the PS4 and I have both my Switch and PS4 hooked up to a capture card, I pressed down the home button to turn off PS4 while it was running and I had Client-RS running, and it woke my Switch from sleep? I'm not kidding. It's done it a bunch of other times too but sometimes it doesn't work. Is that supposed to happen?
